From c6db274aa28373bdae70b9fa974657e6f60bdbb2 Mon Sep 17 00:00:00 2001
From: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Date: Thu, 22 Jan 2015 15:35:16 -0800
Subject: [PATCH 1/1] iio: inv: fix interrupt pin config

currently interrupt pin config uses client->irq value as
default value. But this approach will have issues when
the irq value is not zero or when we use gpio irq. This
patch adds a new param in chip config for pin config and
adds a default mpu6050 pin setting.

Signed-off-by: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
---
 drivers/iio/imu/inv_mpu6050/inv_mpu_core.c | 12 +++++++-----
 drivers/iio/imu/inv_mpu6050/inv_mpu_iio.h  | 10 +++++++++-
 2 files changed, 16 insertions(+), 6 deletions(-)

Index: linux/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
===================================================================
--- linux.orig/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c	2015-02-02 11:44:05.363343946 -0800
+++ linux/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c	2015-02-02 11:52:10.321437873 -0800
@@ -73,6 +73,8 @@
 	.gyro_fifo_enable = false,
 	.accl_fifo_enable = false,
 	.accl_fs = INV_MPU6050_FS_02G,
+	.int_pin_cfg =  INV_MPU6050_BIT_INT_LATCH_EN |
+				INV_MPU6050_BIT_INT_ANYRD_2CLEAR,
 };
 
 static const struct inv_mpu6050_hw hw_info[INV_NUM_PARTS] = {
@@ -646,11 +648,11 @@
 
 	if (enable)
 		ret = inv_mpu6050_write_reg(st, st->reg->int_pin_cfg,
-					st->client->irq |
+					st->chip_config.int_pin_cfg |
 						INV_MPU6050_BIT_BYPASS_EN);
 	else
 		ret = inv_mpu6050_write_reg(st, st->reg->int_pin_cfg,
-					st->client->irq);
+					st->chip_config.int_pin_cfg);
 	return ret;
 }
 EXPORT_SYMBOL_GPL(inv_set_bypass_status);
Index: linux/drivers/iio/imu/inv_mpu6050/inv_mpu_iio.h
===================================================================
--- linux.orig/drivers/iio/imu/inv_mpu6050/inv_mpu_iio.h	2015-02-02 11:44:05.363343946 -0800
+++ linux/drivers/iio/imu/inv_mpu6050/inv_mpu_iio.h	2015-02-02 11:44:05.359343896 -0800
@@ -73,6 +73,7 @@
  *  @accl_fifo_enable:	enable accel data output
  *  @gyro_fifo_enable:	enable gyro data output
  *  @fifo_rate:		FIFO update rate.
+ *  @int_pin_cfg:	Default interrupt pin config
  */
 struct inv_mpu6050_chip_config {
 	unsigned int fsr:2;
@@ -82,6 +83,7 @@
 	unsigned int accl_fifo_enable:1;
 	unsigned int gyro_fifo_enable:1;
 	u16 fifo_rate;
+	u8 int_pin_cfg;
 };
 
 /**
@@ -195,7 +197,13 @@
 #define INV_MPU6050_ONE_K_HZ                              1000
 
 #define INV_MPU6050_REG_INT_PIN_CFG		0x37
-#define INV_MPU6050_BIT_BYPASS_EN		0x2
+#define INV_MPU6050_BIT_ACTL_LOW		0x80
+#define INV_MPU6050_BIT_OPEN_DRAIN		0x40
+#define INV_MPU6050_BIT_INT_LATCH_EN		0x20
+#define INV_MPU6050_BIT_INT_ANYRD_2CLEAR	0x10
+#define INV_MPU6050_BIT_ACTL_FSYNC_LOW		0x08
+#define INV_MPU6050_BIT_FSYNC_INT_MODE_EN	0x04
+#define INV_MPU6050_BIT_BYPASS_EN		0x02
 
 #define INV_MPU6050_REG_WHOAMI			0x75
 #define INV_MPU6500_UNIQUE_ID			0x70
